src/form/wrm.jl

```diff
@@ -1,21 +1,38 @@
 export
-    SDPWRMPowerModel, SDPWRMForm
+    SDPWRMPowerModel, SDPWRMForm,
+    SDPDecompPowerModel, SDPDecompForm
 
 ""
 abstract type AbstractWRMForm <: AbstractConicPowerFormulation end
 
 ""
 abstract type SDPWRMForm <: AbstractWRMForm end
 
+""
+abstract type SDPDecompForm <: AbstractWRMForm end
+
 ""
 const SDPWRMPowerModel = GenericPowerModel{SDPWRMForm}
 
+""
+const SDPDecompPowerModel = GenericPowerModel{SDPDecompForm}
+
+struct SDconstraintDecomposition
+    "Each sub-vector consists of bus IDs corresponding to a clique grouping"
+    decomp::Vector{Vector{Int64}}
+    "Adjacency matrix of chordal graph extension"
+    adj::SparseMatrixCSC{Int64, Int64}
+    "`lookup_index[bus_id] --> idx` for mapping between 1:n and bus indices"
+    lookup_index::Dict
+end
+
 ""
 SDPWRMPowerModel(data::Dict{String,Any}; kwargs...) = GenericPowerModel(data, SDPWRMForm; kwargs...)
 
+SDPDecompPowerModel(data::Dict{String,Any}; kwargs...) = GenericPowerModel(data, SDPDecompForm; kwargs...)
 
 ""
-function variable_voltage(pm::GenericPowerModel{T}; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded = true) where T <: AbstractWRMForm
+function variable_voltage(pm::GenericPowerModel{SDPWRMForm}; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded = true)
```

Comment:
This function signature should be, 
```
function variable_voltage(pm::GenericPowerModel{T}; nw::Int=pm.cnw, cnd::Int=pm.ccnd, bounded = true) where T <: SDPWRMForm
```
Otherwise it cannot be extended later on.

Reply:
Good point. You mean `AbstractWRMForm`, right?